lib/dependency_validator.dart

```diff
@@ -60,22 +60,37 @@ void run({
   logger.fine('transformers:\n'
       '${bulletItems(packagesUsedViaTransformers)}\n');
 
-  // Recursively list all Dart files in lib/
+  // Recursively list all Dart and Scss files in lib/
   final publicDartFiles = <File>[]
     ..addAll(listDartFilesIn('lib/', excludedDirs))
     ..addAll(listDartFilesIn('bin/', excludedDirs));
-  logger.fine('public facing dart files:\n'
-      '${bulletItems(publicDartFiles.map((f) => f.path))}\n');
+
+  final publicScssFiles = <File>[]
+    ..addAll(listScssFilesIn('lib/', excludedDirs))
+    ..addAll(listDartFilesIn('bin/', excludedDirs));
+
+  logger
+    ..fine('public facing dart files:\n'
+        '${bulletItems(publicDartFiles.map((f) => f.path))}\n')
+    ..fine('public facing scss files:\n'
+        '${bulletItems(publicScssFiles.map((f) => f.path))}\n');
 
   // Read each file in lib/ and parse the package names from every import and
   // export directive.
   final packagesUsedInPublicFiles = new Set<String>();
   for (final file in publicDartFiles) {
-    final matches = importExportPackageRegex.allMatches(file.readAsStringSync());
+    final matches = importExportDartPackageRegex.allMatches(file.readAsStringSync());
     for (final match in matches) {
       packagesUsedInPublicFiles.add(match.group(2));
     }
   }
+  for (final file in publicScssFiles) {
+    final matches = importExportScssPackageRegex.allMatches(file.readAsStringSync());
+    for (final match in matches) {
+      packagesUsedInPublicFiles.add(match.group(1)); // TODO: why is this 1?
```

Comment:
```
\b(import|export)\s+['"]{1,3}package:([a-zA-Z0-9_]+)\/[^;]+
```

The Dart regex supports both imports and exports via `(import|export)`. This is technically a group even though we don't care about which one matched. This is why line 84 above uses `match.group(2)`.

---

```
\@import\s+['"]{1,3}package:\s*([a-zA-Z0-9_]+)\/[^;]+
```

The SCSS regex only needs to look for `@import` statements, so it doesn't need that first group, which is why this line checks `match.group(1)`.

Reply:
Ahh that makes sense.